pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -458,6 +471,19 @@ public Set<String> getServingInstances() {
     return _segmentStates.getServingInstances();
   }
 
+  @VisibleForTesting
+  int getGroup(String instanceID) {
+    int group = DEFAULT_SERVER_REPLICA_GROUP_OF_BROKER_VIEW;
+    ServerInstance server = _enabledServerStore.get(instanceID);
+    if (server == null) {
+      LOGGER.warn("Failed to find server {} in the enabledServerManager when update segmentsMap for table {}",
```

Comment:
What are the scenarios where this can happen?

Reply:
I cannot provide exact root cause at the moment. I did see this log during my test where a big upsert table have Error segments. It seems that the server was overloaded.

Reply:
Theoretically, the enabledServer and the segment IS | EV is not updated as a TXN.